engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -125,6 +126,8 @@
 	 */
 	private final List<ValidatedValueUnwrapper<?>> validatedValueHandlers;
 
+	private final ClassLoader userClassLoader;
```

Comment:
Same with some other references which hold on to the given class loader, e.g. `XmlMetaDataProvider`.